src/PhpSpec/ObjectBehavior.php

```diff
@@ -31,9 +31,90 @@
  * @method void beConstructedWith(...$arguments)
  * @method void beConstructedThrough($factoryMethod, array $constructorArguments = array())
  * @method void beAnInstanceOf($class)
+ *
  * @method void shouldHaveType($type)
+ * @method void shouldNotHaveType($type)
+ * @method void shouldReturnAnInstanceOf($type)
```

Comment:
@ciaranmcnulty right so. Can you give me a hint then where to add the magic methods for method/property expectations? Because basically all of those calls are delegated to `Subject` and from there to `Matcher` objects no matter if it is an expectation on the described class, one of its methods or properties (if I understand the voodoo correctly).